app/serializers/api/v1/npq/application_synchronization_serializer.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require "jsonapi/serializer/instrumentation"
+
+module Api
+  module V1
+    module NPQ
+      class ApplicationSynchronizationSerializer
+        include JSONAPI::Serializer
+        include JSONAPI::Serializer::Instrumentation
+
+        attributes :lead_provider_approval_status, :id, :participant_outcome_state
+        attribute(:participant_outcome_state) do |object|
+          object.latest_declaration_of_user&.latest_outcome_of_declaration&.state if object.latest_declaration_of_user.present?
```

Comment:
Here we have potential of many `N+1` queries.

Reply:
Yes I know. We don't have any direct association to fetch the outcomes. Have to chain a lot of methods to get it. I have eliminated the serializer and now doing all of the operation in a single query/iteration in hope for improving performance. Also added a comment in the code. It's still not the most efficient way but better than this.